scene/resources/animation.cpp

```diff
@@ -5562,6 +5562,9 @@ bool Animation::_fetch_compressed_by_index(uint32_t p_compressed_track, int p_in
 // Helper math functions for Variant.
 Variant Animation::add_variant(const Variant &a, const Variant &b) {
 	if (a.get_type() != b.get_type()) {
+		if (a.is_num() && b.is_num()) {
+			return real_t(a) + real_t(b);
+		}
```

Comment:
It is probably better to match the type of the first argument, see cast done in `tween_property()`.

Reply:
There would be an inconsistency with already existing code in `interpolate_variant()`. It at least looks to me like the result is converted back to int when needed, anyhow. I could be wrong about it, however.